koku/masu/util/ibm/download.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""IBM Download Utils."""
+import logging
+
+from django.conf import settings
+from ibm_cloud_sdk_core import BaseService
+from ibm_cloud_sdk_core.api_exception import ApiException
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_platform_services.common import get_sdk_headers
+
+from api.provider.models import Provider
+from masu.config import Config
+from masu.util.aws.common import copy_local_report_file_to_s3_bucket
+from masu.util.common import get_path_prefix
+
+LOG = logging.getLogger(__name__)
+
```

Comment:
I am not quite sure if this needs to be in its own utl since these methods would probably only be used in the `ibm_report_downloader.py` file. It may it easier to just have them methods in the `IBMReportDownloader` class, but that is more of a preference comment really.

Reply:
You know. In a second thought we can do it. I just don't want to make the class bloated and just do one thing and that's preparing the csv report files for the processor.

That's why I created `util/ibm/download` so it is easy to separate between the responsibility of the class and the utilities it uses. I guess I can move these to the ibm_report_downloader.py file but not inside the class but as functions that are used by that class. Will that work?

If you still think it should  be inside the class, let me know. Would like to hear more.